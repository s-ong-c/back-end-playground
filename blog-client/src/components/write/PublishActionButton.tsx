import * as React from 'react';
import styled from 'styled-components';
import Button from '../common/Button';
const PublishActionButtonBlock = styled.div`
  display: flex;
  justify-content: flex-end;
`;
const StyledButton = styled(Button)`
  height: 2.5rem;
  padding-left: 1.12rem;
  padding-right: 1.12rem;
  & + & {
    margin-left: 0.875rem;
  }
`;
export interface PublishActionButtonProps {
  onCancel: () => void;
  onPublish: () => void;
}

const PublishActionButton: React.SFC<PublishActionButtonProps> = ({
  onCancel,
  onPublish,
}) => {
  return (
    <PublishActionButtonBlock>
      <StyledButton data-testid="cancelPublish" color="gray" onClick={onCancel}>
        취소
      </StyledButton>
      <StyledButton data-testid="publish" onClick={onPublish}>
        출간하기
      </StyledButton>
    </PublishActionButtonBlock>
  );
};

export default PublishActionButton;
