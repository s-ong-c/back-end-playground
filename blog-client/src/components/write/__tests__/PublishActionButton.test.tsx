import * as React from 'react';
import { render, fireEvent } from 'react-testing-library';
import PublishActionButton, {
  PublishActionButtonProps,
} from '../PublishActionButton';

describe('PublishActionButton', () => {
  const setup = (props: Partial<PublishActionButtonProps> = {}) => {
    const initialProps: PublishActionButtonProps = {
      onCancel: () => {},
      onPublish: () => {},
    };
    const utils = render(<PublishActionButton {...initialProps} {...props} />);
    const buttons = {
      cancel: utils.getByText('취소'),
      publish: utils.getByText('출간하기'),
    };
    return {
      ...utils,
      buttons,
    };
  };
  it('renders property', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('calls onCancel and onPublish', () => {
    const onCancel = jest.fn();
    const onPublish = jest.fn();
    const utils = setup({
      onCancel,
      onPublish,
    });
    fireEvent.click(utils.buttons.publish);
    expect(onPublish).toBeCalled();
    fireEvent.click(utils.buttons.cancel);
    expect(onCancel).toBeCalled();
  });
});
