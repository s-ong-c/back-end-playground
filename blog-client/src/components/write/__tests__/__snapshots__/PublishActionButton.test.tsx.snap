// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishActionButton matches snapshot 1`] = `
<div>
  <div
    class="sc-bwzfXH eprlzE"
  >
    <button
      class="sc-bdVaJa bAArFn sc-htpNat kIspaC"
      color="gray"
      data-testid="cancelPublish"
    >
      취소
    </button>
    <button
      class="sc-bdVaJa lgjZOb sc-htpNat kIspaC"
      color="teal"
      data-testid="publish"
    >
      출간하기
    </button>
  </div>
</div>
`;
